doc/source/dev/create_release.rst

```diff
@@ -1,3 +1,14 @@
+Creating Galaxy Releases
+========================
+
+The "main" release process is an interactive checklist with instructions (see `Publication of Galaxy Release v 23.2 https://github.com/galaxyproject/galaxy/issues/16742>`_ for an example).
+This issue is generated via `make release-issue` at the end of the process.
```

Comment:
```suggestion
This issue is generated via `make release-issue`.
```
I think this is all we need here? I think "at the end of the process" may be a little confusing since we generate it at the beginning of the release process (and the last bullet points in the issue refer to the *next* release).